fix for metrics

bodintsov opened the pull request:
## Purpose

Fix to enable API serializer to combine preprint metrics for Versioned GUID and Base GUID

## Changes

Changed the logic of the method for searching metrics.

## QA Notes

TBD

## Documentation

TBD

## Side Effects

TBD

## Ticket

https://openscience.atlassian.net/browse/ENG-7086

brianjgeiger reviewed (commented):
Is this going to both show the preprint counts from before the changeover, plus the views/downloads after we split things into versions? Or will it only show the counts for things that happened from before the change to versions?

bodintsov commented:
> Is this going to both show the preprint counts from before the changeover, plus the views/downloads after we split things into versions? Or will it only show the counts for things that happened from before the change to versions?

I'm not sure, as it deals with counting and I just changed how these numbers are selected (in the ticket description, as Matt mentioned). If any concerns I will research this question.

brianjgeiger commented:
I am 80% sure you'll only get the old statistics with this, and you need to do a combo of both in order to fulfill the ticket requirements. Yes, please research.

bodintsov commented:
I checked for proper counts of downloads/views, and it works as expected. Current implementation sums preprint metrics counts from before the changeover and after.
